Services/DataBaseService/Repositories/TradeRepository.cs

```diff
@@ -0,0 +1,172 @@
+        private readonly ILogger<TradeRepository> logger;
+
+        public TradeRepository(
+            ITradeMapper mapper, TPlatformDbContext dbContext, 
+            [FromServices] ILogger<TradeRepository> logger
+            )
+        {
+            this.mapper = mapper;
+            this.dbContext = dbContext;
+            this.logger = logger;
+        }
+
+        private DbUserBalance RegisterUserBalance(Guid userId)
+        {
+            var userBalance = new UserBalance()
+            {
+                Id = Guid.NewGuid(),
+                UserId = userId,
+                BalanceInRub = 0,
+                BalanceInUsd = 0,
+                BalanceInEur = 0
+            };
+            var dbUserBalance = mapper.MapToDbUserBalance(userBalance);
+            dbContext.UserBalances.Add(dbUserBalance);
+            dbContext.SaveChanges();
+            logger.LogInformation("New user balance added");
+            return dbUserBalance;
+        }
+
+        private void UpdateBalanceAfterTransaction(Transaction transaction)
+        {
+            var dbUserBalance = dbContext.UserBalances.FirstOrDefault(
+                user => user.UserId == transaction.UserId);
+
+            var sign = transaction.Operation == OperationType.Sell ? 1 : -1;
+            var cost = sign * transaction.Count * transaction.Price;
+
+            switch (transaction.Currency)
+            {
+                case Currency.Rub:
+                    dbUserBalance.BalanceInRub += cost;
+                    break;
+                case Currency.Usd:
+                    dbUserBalance.BalanceInUsd += cost;
+                    break;
+                case Currency.Eur:
+                    dbUserBalance.BalanceInEur += cost;
+                    break;
+                default:
+                    var exception = new BadRequestException("Currency isn't correct");
+                    logger.LogWarning(exception, "Request with currency isn't correct");
+                    throw exception;
+            }
+            logger.LogInformation("Request to update balance of user {transaction.UserId} finished successfully");
+        }
+
+        private void UpdatePortfolioAfterTransaction(Transaction transaction)
+        {
+            var dbPortfolio = dbContext.Portfolios.FirstOrDefault(
+                p => p.Figi == transaction.Figi && p.UserId == transaction.UserId);
+            
+            if(dbPortfolio == null && transaction.Operation == OperationType.Buy)
+            {
+                dbPortfolio = new DbPortfolio()
+                {
+                    Figi = transaction.Figi,
+                    Count = transaction.Count,
+                    UserId = transaction.UserId,
+                    Broker = transaction.Broker.ToString()
+                };
+                dbContext.Portfolios.Add(dbPortfolio);
+                logger.LogInformation("Request to add new instrument{transaction.Figi}" +
```

Comment:
Isn't it necessary to use interpolation - $ before the string here?

Reply:
Yes, thanks